docs/building-open-mpi/required-support-libraries.rst

```diff
@@ -0,0 +1,128 @@
+.. _install-required-support-libraries-label:
+
+Required support libraries
+--------------------------
+
+Open MPI uses the following support libraries:
+
+* `Hardware Locality (hwloc) <https://www.open-mpi.org/projects/hwloc/>`_
+* `Libevent <https://libevent.org/>`_
+* `PMIx <https://pmix.org/>`_
+* `PRRTE <https://github.com/openpmix/prrte>`_
+
+Full copies of the source code of these libraries are bundled in the
+Open MPI source code distribution for end-user convenience.
+
+* In some cases, the presence of these bundled versions can lead to
+  obscure corner cases.  If this has happened to you, continue reading
+  this section.
+* If you have no problems with Open MPI and these four support
+  libraries, then you can safely skip reading this section.
+
+Bundled versions
+^^^^^^^^^^^^^^^^
+
+These libraries are bundled in Open MPI because they are *required*
+for correct Open MPI functionality (with the exception of PRRTE, which
+can be omitted if another PMIx-compatible run-time system is
+available, such as SLURM), and are not universally available by
+default on many systems.  For example, macOS does not include
+any of these libraries.
+
+Rather than make end-users separately download and install these
+required libraries, Open MPI bundles them in its distribution tarball,
+and will automatically use the bundled versions if compatible versions
+cannot be found installed on the build system.
+
+System-provided versions
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+That being said, many systems *do* provide installed versions of these
```

Comment:
We should really encourage users to install these libraries from their os distro repos, rather than use the internal versions.  Life is better if we're not installing libhwloc into random locations.